src/app/saga.ts

```diff
@@ -279,8 +280,17 @@ export function* handleDeepLink(action: OpenDeepLink) {
 
   const rawParams = parse(deepLink)
   if (rawParams.path) {
-    const pathParts = rawParams.path.split('/')
-    if (rawParams.path.startsWith('/v/')) {
+    // handle dynamic link
+    if (rawParams.hostname === 'vlra.app') {
+      const dynamicLink = yield call(resolveDynamicLink, rawParams.href)
```

Comment:
Was this tested on both iOS and Android?

I _believe_ `deepLink` here should already be the resolved link, not the short link (i.e., it should be `vlra.app/share/:address`, not `vlra.app/:shareLinkId`). But it likely depends on which way the link was grabbed in `App.js`.

This wasn't really an issue when we were using long links, because it was always going to be the same link. But now that we're using short links, it may cause issues.

Reply:
I'm now realizing that you changed the domain of the URL when creating the link - so the issue I mentioned is likely to not actually be an issue, since anything coming from `vlra.app` should always be the _un-_resolved link.

Reply:
hmm yeah good shout, it's been a while since i tested this and am only 99% sure i tested on both iOS and Android. i'll test again tomorrow before i merge to make sure :) i did change the domain of the `link`, and added a redirect on our website so that now if you click on a share link like `https://vlra.app/vqT5MqgEVCVnBjJB7` on desktop, you land on the valora home page

Reply:
OK I'm testing this on Android and it's not working properly, per my initial concern. The link being passed into `handleDeepLink` is `https://valoraapp.com/share/:address` - so the `rawParams.hostname === 'vlra.app'` check fails.

We probably need to tweak the platform-specific stuff in `App.tsx` to make sure the same link (pre-resolved, i.e. `vlra.app`; or post-resolved, i.e. `valoraapp.com`) gets passed into `handleDeepLink`. This is likely because the link is coming from difference libraries depending on the platform (`CleverTap` for Android, `dynamicLinks` for iOS).

Let me know if you want to chat or pair on this, I'm happy to jump on a call.

Reply:
thanks for picking this up @jacob-israel-turner ! i think i've fixed it, if you want to take another look. i tested generating and consuming the links on iOS and android again and seems to be working end to end :D